### ISSUE TITLE
`UnnecessaryFullyQualifiedName` false positive

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#9466]
## Expected Behavior
This code shouldn't report any Issue because I have an import for the `Foo` inside `foo.bar1` but I can't import the one from `foo.bar2`. So the FullyQualified of `foo.bar2` are needed.

```kotlin
import foo.bar1.Foo

fun main() {
    Foo()
    foo.bar2.Foo(
        foo.bar2.Foo.Color("dark")
    )
    foo.bar2.Foo(
        foo.bar2.Foo.Color.white
    )
}
```

## Observed Behavior
I get 4 issues (one for each fully qualified name)

## Steps to Reproduce
You can reproduce this issue with this Unit test:

```kotlin
        @Test
        fun `does not report when there is another import`() {
            val code = """
                import foo.bar1.Foo

                fun main() {
                    Foo()
                    foo.bar2.Foo(
                        foo.bar2.Foo.Color("dark")
                    )
                    foo.bar2.Foo(
                        foo.bar2.Foo.Color.white
                    )
                }
            """.trimIndent()
            val annotationCode = """
                package foo.bar1
                fun Foo() {}
            """.trimIndent()
            val interfaceCode = """
                package foo.bar2
                data class Foo(
                    val color: Color
                ) {
                    class Color(name: String) {
                        companion object {
                            val white = Color("white")
                        }
                    }
                }
            """.trimIndent()

            assertThat(subject.lintWithContext(env, code, annotationCode, interfaceCode)).isEmpty()
        }
```

## Your Environment
* Version of detekt used: main

**Comment by BraisGabin** [comment]
I don't have the time to dig enough in this issue. I find out that the problem is in `findLocalSymbols` that doesn't say that `Foo` is already a LocalSymbol (because it is imported).

**Comment by mwaddo** [comment]
Hi, I confirmed the problem is in [findLocalSymbols](https://github.com/detekt/detekt/blob/ff22c8aff2b5f14acd3a341f2b3d99a33189c117/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UnnecessaryFullyQualifiedName.kt#L254-L264) as you suspected. It only looks for class naming conflicts, but not functions since it only queries `scope.classifiers(name)`. So `fun Foo()` being imported is invisible and the collision is undetected.

There's a second bug from what I can tell too. Or at least, in order to get 0 findings in the test you provided, another change is required. Even after the aforementioned fix, [hasNameCollision](https://github.com/detekt/detekt/blob/ff22c8aff2b5f14acd3a341f2b3d99a33189c117/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UnnecessaryFullyQualifiedName.kt#L234-L252) only checks the immediate symbol (`Color`), not its outer class (`Foo`). So `foo.bar2.Foo.Color(...)` still gets flagged even though `Foo` is blocked.

I'll put out a PR addressing both
